pkg/build/build.go

```diff
@@ -73,14 +74,24 @@ func Create(s *stack.Stack, t *target.Target) error {
 }
 
 // CreateBaseDev builds images for code-as-config
-func CreateBaseDev(stackPath string, imagesToBuild map[string]string) error {
+func CreateBaseDev(s *stack.Stack) error {
 	ce, err := containerengine.Discover()
 	if err != nil {
 		return err
 	}
+	imagesToBuild := map[string]string{}
+	for _, f := range s.Functions {
+		rt, err := utils.NewRunTimeFromFilename(f.Handler)
+		if err != nil {
+			return err
+		}
+		lang := rt.String()
+		_, ok := imagesToBuild[lang]
+		if ok {
+			continue
+		}
 
-	for lang, imageTag := range imagesToBuild {
-		f, err := os.CreateTemp(stackPath, fmt.Sprintf("%s.*.dockerfile", lang))
+		f, err := os.CreateTemp(s.Path(), fmt.Sprintf("%s.*.dockerfile", lang))
```

Comment:
The stackPath contains nitric.yaml, so nitric.yaml is being treated as a directory in this case.